src/Bicep.Core/Syntax/AssertDeclarationSyntax.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Bicep.Core.Navigation;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public class AssertDeclarationSyntax : StatementSyntax, ITopLevelNamedDeclarationSyntax
+    {
+        public AssertDeclarationSyntax(IEnumerable<SyntaxBase> leadingNodes, Token keyword, IdentifierSyntax name, SyntaxBase assignment, SyntaxBase value)
+            : base(leadingNodes)
+        {
+            AssertKeyword(keyword, nameof(keyword), LanguageConstants.AssertKeyword);
+            AssertSyntaxType(name, nameof(name), typeof(IdentifierSyntax));
```

Comment:
```suggestion
```

No need for an assertion here -- the `name` constructor parameter already has a type of `IdentifierSyntax`.